packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -123,19 +177,92 @@ class DatePickerInput extends React.Component {
     }
   }
 
-  handleOnChange(event) {
-    if (!DateUtil.validDateInput(event.target.value)) {
-      return;
+  handleOnKeyDown(event) {
+    if (this.props.onKeyDown) {
+      this.props.onKeyDown(event);
     }
+  }
 
-    if (this.props.onChange) {
-      this.props.onChange(event);
+  handleDayChange(event) {
+    const inputValue = event.target.value.replace(/\D/gm, '');
+    this.handleDateChange(event, DateUtil.inputType.DAY, inputValue);
+  }
+
+  handleMonthChange(event) {
+    const inputValue = event.target.value.replace(/\D/gm, '');
+    this.handleDateChange(event, DateUtil.inputType.MONTH, inputValue);
+  }
+
+  handleYearChange(event) {
+    const inputValue = event.target.value.replace(/\D/gm, '');
+    this.handleDateChange(event, DateUtil.inputType.YEAR, inputValue);
```

Comment:
nit: Since `handleDayChange`, `handleMonthChange`, and `handleYearChange` all do the same except for the type passed to `handleDateChange`, the line to get the inputValue from the event can be moved to `handleDateChange`.

Reply:
Changed! [0b496eb](https://github.com/cerner/terra-framework/pull/1004/commits/0b496eb729d1b0fc5851cc50d820b24a267a67ec)